cmd/kubelet/app/server.go

```diff
@@ -425,7 +425,7 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 			if err != nil {
 				return err
 			}
-			if cloud == nil {
+			if s.CloudProvider != "external" && cloud == nil {
```

Comment:
for followup, shouldn't we skip this whole block if "external", starting from the `if` at L423 ?